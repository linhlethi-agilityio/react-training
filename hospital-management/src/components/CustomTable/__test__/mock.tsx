import { ITableColumn } from '@components/CustomTable/types';
import { IData } from '@type/data';
import { Avatar } from '@mantine/core';

export interface IDataType extends IData {
  id: string;
  avatar: string;
  doctor: string;
}

export const columnsConfig: ITableColumn<IDataType>[] = [
  {
    id: '1',
    width: '232px',
    header: 'Doctor',
    accessor: 'doctor',
  },
  {
    id: '2',
    width: '80px',
    header: 'Avatar',
    accessor: (data: IDataType) => <Avatar src={data.avatar} />,
  },
];

export const data: IDataType[] = [
  {
    id: '1',
    avatar: 'https://images.vexels.com/content/145908/preview/male-avatar-maker-2a7919.png',
    doctor: 'Dr. Theresa Webb',
  },
  {
    id: '2',
    avatar: 'https://images.vexels.com/content/145908/preview/male-avatar-maker-2a7919.png',
    doctor: 'Dr. Arlene McCoy',
  },
];
