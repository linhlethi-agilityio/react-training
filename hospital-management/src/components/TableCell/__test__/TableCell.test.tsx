import { describe, expect, it } from 'vitest';
import { MemoizedTableCell } from '..';
import { render } from '@testing-library/react';
import { columnsConfig, data } from './mock';

describe('testing TableCell component', () => {
  it('Snapshot typeof accessor string', () => {
    const tableCell = render(<MemoizedTableCell item={data} columnConfig={columnsConfig[0]} />);

    expect(tableCell).toMatchSnapshot();
  });

  it('Snapshot typeof accessor function', () => {
    const tableCell = render(<MemoizedTableCell item={data} columnConfig={columnsConfig[1]} />);

    expect(tableCell).toMatchSnapshot();
  });

  it('should render the cell value correct', () => {
    const { getByText } = render(<MemoizedTableCell item={data} columnConfig={columnsConfig[0]} />);

    expect(getByText(data.doctor)).toBeTruthy();
  });
});
