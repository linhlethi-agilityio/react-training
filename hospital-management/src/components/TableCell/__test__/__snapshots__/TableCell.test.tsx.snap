// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`testing TableCell component > Snapshot typeof accessor function 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <td>
        <div
          class="mantine-Avatar-root mantine-1rdmhww"
        >
          <img
            class="mantine-9rx0rd mantine-Avatar-image"
            src="https://images.vexels.com/content/145908/preview/male-avatar-maker-2a7919.png"
          />
        </div>
      </td>
    </div>
  </body>,
  "container": <div>
    <td>
      <div
        class="mantine-Avatar-root mantine-1rdmhww"
      >
        <img
          class="mantine-9rx0rd mantine-Avatar-image"
          src="https://images.vexels.com/content/145908/preview/male-avatar-maker-2a7919.png"
        />
      </div>
    </td>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing TableCell component > Snapshot typeof accessor string 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <td>
        Dr. Theresa Webb
      </td>
    </div>
  </body>,
  "container": <div>
    <td>
      Dr. Theresa Webb
    </td>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
